### PR TITLE
Bump up some library versions.

### PR DESCRIPTION
**Pull request by thagikura**
Also fixes:
 - Some Kotlin errors in the demo app.
 - Replace the deprecated compile dependency syntax to `api` or `implementation` in the build.gradle files.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -19,14 +19,14 @@
 buildscript {
     ext {
         minSdkVersion = 14
-        targetSdkVersion = 26
-        compileSdkVersion = 26
-        buildToolsVersion = "26.0.0"
+        targetSdkVersion = 27
+        compileSdkVersion = 27
+        buildToolsVersion = "27.0.0"
```
Comment (MGaetan89) [inline build.gradle]: `27.0.1` is the latest version

File: build.gradle [hunk 2]
```diff
@@ -19,14 +19,14 @@
...
         androidGradlePluginVersion = "2.3.3"
         androidMavenGradlePluginVersion = "1.5"
         gradleBintrayPluginVersion = "1.6"
-        kotlinVersion = "1.1.3"
-        supportLibVersion = "26.0.0"
+        kotlinVersion = "1.1.60"
```
Comment (MGaetan89) [inline build.gradle]: `1.1.61` is available

File: build.gradle [hunk 3]
```diff
@@ -19,14 +19,14 @@
...
+        supportLibVersion = "27.0.0"
```
Comment (MGaetan89) [inline build.gradle]: `27.0.1` is available

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,6 +1,6 @@
-#Tue Jun 13 19:10:11 JST 2017
+#Mon Nov 27 14:52:10 JST 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-all.zip
```
Comment (MGaetan89) [inline gradle/wrapper/gradle-wrapper.properties]: Why not go straight to `4.3.1`?

File: demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt [hunk 5]
```diff
@@ -248,8 +248,8 @@ internal class FlexItemEditFragment : DialogFragment() {
                     if (i + 1 < textViews.size) {
                         textViews[i + 1].requestFocus()
                     } else if (i == textViews.size - 1) {
-                        val inputMethodManager = activity
-                                .getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+                        val inputMethodManager = activity?.getSystemService(
+                                Context.INPUT_METHOD_SERVICE) as InputMethodManager
                         inputMethodManager.hideSoftInputFromWindow(v.windowToken, 0)
```
Comment (MGaetan89) [inline demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt]: If `activity` is `null`, then `inputMethodManager` is also `null`. But there is no safe-call here.

File: build.gradle (hunk 1 above)
Comment (thagikura) [reply build.gradle]: Done

File: build.gradle (hunk 2 above)
Comment (thagikura) [reply build.gradle]: Done

File: build.gradle (hunk 3 above)
Comment (thagikura) [reply build.gradle]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt (hunk 5 above)
Comment (thagikura) [reply demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt]: Done

File: gradle/wrapper/gradle-wrapper.properties (hunk 4 above)
Comment (thagikura) [reply gradle/wrapper/gradle-wrapper.properties]: Done

File: circle.yml [hunk 6]
```diff
@@ -6,7 +6,7 @@ machine:
 
 dependencies:
     pre:
-        - echo y | android update sdk --no-ui --all --filter "android-26,build-tools-26.0.0,tools,platform-tools,extra-android-m2repository"
+        - echo y | android update sdk --no-ui --all --filter "android-27,build-tools-27.0.0,tools,platform-tools,extra-android-m2repository"
```
Comment (MGaetan89) [inline circle.yml]: Build tools needs to be updated to `27.0.1` here too

Comment (thagikura) [reply circle.yml]: Oh that's right. Thanks for catching it.

**Comment by MGaetan89** [comment]
Kotlin 1.2.0 is out now, if you want to update right away. Otherwise it looks good to me 👍

**Comment by thagikura** [comment]
Thanks. I'll merge this once CI becomes green.

**Pull request merged**